lib/climate_control/modifier.rb

```diff
@@ -33,10 +33,24 @@ def run_block
       @block.call
     end
 
+    def stringify_boolean(value)
+      return value.to_s if [true, false].include?(value)
+      value
+    end
+
+    def stringify_number(value)
+      return value.to_s if value.is_a?(Numeric)
+      value
+    end
+
+    def stringify_value(value)
+      stringify_number(stringify_boolean(value))
+    end
+
     def copy_overrides_to_environment
       @environment_overrides.each do |key, value|
         begin
-          @env[key] = value
+          @env[key] = stringify_value(value)
```

Comment:
Yeah, I'd love to use `value.to_s` here but ended up adding the additional methods because the "raises when the value cannot be assigned properly" spec fails if `value.to_s` is used. I was leery of modifying that spec in case there was some history/special situation there that I didn't understand. Here is the stack dump in case it helps:

```
       expected ClimateControl::UnassignableValueError with message matching /attempted to assign .*Thing.* to FOO but failed \(no implicit conversion of Thing into String\)$/, got #<NoMethodError: undefined method `call' for nil:NilClass> with backtrace:
         # ./lib/climate_control/modifier.rb:33:in `run_block'
         # ./lib/climate_control/modifier.rb:15:in `block in process'
         # ./lib/climate_control/environment.rb:11:in `synchronize'
         # ./lib/climate_control/environment.rb:11:in `synchronize'
         # ./lib/climate_control/modifier.rb:12:in `process'
         # ./lib/climate_control.rb:10:in `modify'
         # ./spec/acceptance/climate_control_spec.rb:134:in `with_modified_env'
         # ./spec/acceptance/climate_control_spec.rb:129:in `block (3 levels) in <top (required)>'
         # ./spec/acceptance/climate_control_spec.rb:128:in `block (2 levels) in <top (required)>'
         # ./spec/acceptance/climate_control_spec.rb:152:in `block (2 levels) in <top (required)>'
```

That seems like a valid situation that the `value.to_s` implementation wouldn't handle probably.

Happy to do whatever makes sense, though.